infrastructure/lib/atat-web-api-stack.ts

```diff
@@ -34,125 +34,91 @@ export class AtatWebApiStack extends cdk.Stack {
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/getPortfolioDrafts.ts",
     });
 
-    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraft", {
-      resource: portfolioDraftId,
+    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraftFunction", {
       table: table,
       method: HttpMethod.DELETE,
       handlerPath: packageRoot() + "/api/portfolioDrafts/deletePortfolioDraft.ts",
     });
 
-    const createPortfolioStep = new ApiDynamoDBFunction(this, "CreatePortfolioStep", {
-      resource: portfolio,
-      table: table,
-      method: HttpMethod.POST,
-      handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/createPortfolioStep.ts",
-    });
-
-    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStep", {
-      resource: portfolio,
+    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/getPortfolioStep.ts",
     });
 
-    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStep", {
-      resource: funding,
+    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStepFunction", {
       table: table,
       method: HttpMethod.POST,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/createFundingStep.ts",
     });
 
-    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStep", {
-      resource: funding,
+    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/getFundingStep.ts",
     });
 
+    // The API spec, which just so happens to be a valid CloudFormation snippet (with some actual CloudFormation
+    // in it) gets uploaded to S3
+    const apiAsset = new s3asset.Asset(this, "ApiSpecAsset", {
+      path: "./atat_provisioning_wizard_api.yaml",
+    });
+
+    // And now we include that snippet as an actual part of the template using the AWS::Include Transform. This
+    // is where the two previous pieces come together. Now, all the Fn::Sub and other functions will be resolved
+    // which means that we can reference the ARNs of the various functions.
+    const data = cdk.Fn.transform("AWS::Include", { Location: apiAsset.s3ObjectUrl });
+
+    // And with the data now loaded from the template, we can use ApiDefinition.fromInline to parse it as real
+    // OpenAPI spec (because it was!) and now we've got all our special AWS values and variables interpolated.
+    const restApi = new apigw.SpecRestApi(this, "AtatSpecTest", {
+      apiDefinition: apigw.ApiDefinition.fromInline(data),
+      endpointTypes: [apigw.EndpointType.REGIONAL],
+      parameters: {
+        endpointConfigurationTypes: apigw.EndpointType.REGIONAL,
+      },
```

Comment:
We really ought to have a comment for why it's necessary to be repetitive here. And the answer to that is that when you're actually loading in these values from a `SpecRestApi`, API Gateway ignores the `endpointTypes` field in favor of `parameters.endpointConfigurationTypes`. We should first validate this is actually still a required workaround. If it is, then it should get a clear comment so that someone helpful trying to clean up duplication (probably future me or future you) doesn't remove this and break things. It will also be helpful when this needs to change to `.PRIVATE`.